### PR TITLE
.maintain/monitoring/alerting-rules: Adjust transaction queue size alert

### PR DESCRIPTION
**Pull request by mxinden**
The transaction queue size alert has been firing with a constant 10
transactions in the queue. While maybe problematic those 10 transactions
don't need to be the same across scrape intervals.

Instead of alerting with a size above 10, alert based on two things:

1. Monotonically increasing queue size

2. Upper limit queue size reached

### CODE REVIEW

File: .maintain/monitoring/alerting-rules/alerting-rules.yaml [hunk 1]
```diff
@@ -73,24 +73,33 @@ groups:
...
     labels:
       severity: critical
     annotations:
-      message: 'The node {{ $labels.instance }} has more than 10 transactions in
-      the queue for more than 30 minutes'
+      message: 'The transaction pool size on node {{ $labels.instance }} has
+      been monotonically increasing for the last 30 minutes.'
+  - alert: TransactionQueueSizeHigh
+    expr: 'polkadot_sub_txpool_validations_scheduled -
+    polkadot_sub_txpool_validations_finished > 10000'
```
Comment (mxinden) [inline .maintain/monitoring/alerting-rules/alerting-rules.yaml:96]: `> 10_000` is a rather loose requirement. As of today a queue size `> 8_000` for a couple of minutes is not an infrequence. I hope we can tighten this limit in the future.

**Comment by mxinden** [comment]
//CC @fgimenez

File: .maintain/monitoring/alerting-rules/alerting-rules.yaml [hunk 2]
```diff
@@ -73,24 +73,33 @@ groups:
   # Transaction queue
   ##############################################################################
 
-  - alert: TransactionQueueSize
-    expr: 'polkadot_sub_txpool_validations_scheduled -
-    polkadot_sub_txpool_validations_finished > 10'
+  - alert: TransactionQueueSizeIncreasing
+    expr: 'increase(polkadot_sub_txpool_validations_scheduled[5m]) -
```
Comment (fgimenez) [inline .maintain/monitoring/alerting-rules/alerting-rules.yaml:77]: Should `polkadot_sub_txpool_validations_invalid` be taken into account too? Maybe `increase(polkadot_sub_txpool_validations_scheduled[5m]) - increase(polkadot_sub_txpool_validations_finished[5m]) - increase(polkadot_sub_txpool_validations_invalid[5m]) > 0` WDYT?

File: .maintain/monitoring/alerting-rules/alerting-rules.yaml [hunk 3]
```diff
@@ -73,24 +73,33 @@ groups:
...
     annotations:
-      message: 'The node {{ $labels.instance }} has more than 10 transactions in
-      the queue for more than 10 minutes'
-  - alert: TransactionQueueSize
-    expr: 'polkadot_sub_txpool_validations_scheduled -
-    polkadot_sub_txpool_validations_finished > 10'
+      message: 'The transaction pool size on node {{ $labels.instance }} has
+      been monotonically increasing for the last 10 minutes.'
+  - alert: TransactionQueueSizeIncreasing
+    expr: 'increase(polkadot_sub_txpool_validations_scheduled[5m]) -
```
Comment (fgimenez) [inline .maintain/monitoring/alerting-rules/alerting-rules.yaml:86]: Same as above about `polkadot_sub_txpool_validations_invalid`

File: .maintain/monitoring/alerting-rules/alerting-rules.yaml [hunk 4]
```diff
@@ -73,24 +73,33 @@ groups:
...
     for: 30m
     labels:
       severity: critical
     annotations:
-      message: 'The node {{ $labels.instance }} has more than 10 transactions in
-      the queue for more than 30 minutes'
+      message: 'The transaction pool size on node {{ $labels.instance }} has
+      been monotonically increasing for the last 30 minutes.'
+  - alert: TransactionQueueSizeHigh
+    expr: 'polkadot_sub_txpool_validations_scheduled -
```
Comment (fgimenez) [inline .maintain/monitoring/alerting-rules/alerting-rules.yaml:95]: Same as above about `polkadot_sub_txpool_validations_invalid`, in this case `polkadot_sub_txpool_validations_scheduled - polkadot_sub_txpool_validations_finished - polkadot_sub_txpool_validations_invalid > 10000`

**Comment by mxinden** [comment]
@fgimenez as far as I can tell `polkadot_sub_txpool_validations_finished` includes all validations detached from validation result. @NikVolf (who implemented them) can you confirm?

https://github.com/paritytech/substrate/blob/c8c16d1803c5d84565ffb788ab35e2cedf289530/client/transaction-pool/src/lib.rs#L271-L277

**Comment by mxinden** [comment]
@gabreal @ddorgan @tomaka could one of you approve this pull request in case you have no objections?

**Pull request merged**